### DISCUSSION TITLE
Unable to server files from web_root, it's returning 404

### DISCUSSION THREAD

**Discussion by mslugx** [cesanta/mongoose#2526] [Ask questions, get answers] [answered]
Hi All:

    I am trying to enable to serve front end files from my local environment according to this screenshot here. I think it's possible right? 

![image](https://github.com/cesanta/mongoose/assets/15797917/475b4cf6-02af-4f4a-801d-b1cc099a23da)
I have the files including index.html stored under web_root directory. Here is my code just like the example. 

![image](https://github.com/cesanta/mongoose/assets/15797917/6d79071a-419b-4704-b091-523a3cca13bf)

However when i trying to access. http://ipaddress/   the board returned 404, and i tried http://ipaddress/index.html it's returning 404 as well. I tried different combination of  MG_ENABLE_PACKED_FS and MG_ENABLE_FILE It doesn't seems to make any different. I am using the bare metal Architecture. Can someone point me to the correct direction? Thanks.

**Comment by scaprile** [comment, marked as answer]
We already know our tutorials, please don't paste them here, just refer to them
Our examples work, so please be run the example _as is_ and learn from it. Do not try, do as the tutorial says and as you can see in the Makefile.
If there is an actual issue, please open an issue, _honor the issue template_ so we can know your scenario and reproduce the issue if necessary, and let us know _exactly_ what you are doing
What is "your local environment" ? What is "the board" ?
What is your filesystem ? Do you actually have one ?
If you want to serve from a filesystem, then don't define the packed file system and serve from your file system, make sure it is supported and configure for it. The example configures for regular POSIX filesystems
If you want to serve from a packed filesystem, you have to make it. Did you follow the process to pack your files ? Mongoose will serve what is in packed_fs.c
If you are running in baremetal then my guess is that you don't have a filesystem, how do you expect Mongoose to find your files ? How do you expect your files to end up in "your board" in the first place ?
Please follow the guidelines in the [documentation](https://mongoose.ws/documentation),  [examples](https://github.com/cesanta/mongoose/tree/master/examples) and [tutorials](https://mongoose.ws/tutorials/) available. Understand them. Know your tools, know your development environment, understand what you are doing.

**Comment by mslugx** [comment]
Hi Scaprile:

    My apology for the wrong template. I will follow the template in the future. It's not really an issue of the library, it's just i am not sure why it's not working, so I want to ask you guys for some guidance. I stuck on this issue for a couple days. 

Thank you for your pointer about bare metal. I think i was looking at an older version of the example by clicking on the link of " view code on github", however the code was already updated. Now the code is working with only Linux or Windows instead of the flag "MG_ENABLE_PACKED_FS".
